src/main/webapp/mapScript.js

```diff
@@ -25,21 +34,86 @@ function initMap() {
 function geocodeAddress(geocoder, resultsMap) {
     const address = document.getElementById('address').value;
     geocoder.geocode({'address': address}, (results, status) => {
-    if (status === 'OK') {
-        resultsMap.setCenter(results[0].geometry.location);
+        if (status === 'OK') {
+            resultsMap.setCenter(results[0].geometry.location);
+        } else {
+            alert('Geocode was not successful for the following reason: ' + status);
+        }
+    });
+}
+
+/**
+ * Determines which group was checked or unchecked
+ * Then sends that group to either be removed from the map if unchecked
+ * Or added to the map if checked
+ */
+function addGroupChecked(checkbox, groupName) {
```

Comment:
Re: groupName
It's safer to pass the groupId around for getting/setting group info for the following reasons:
  + There could be multiple groups with the same name
  + Future-proofness: Maybe in the future you'll allow users to update the group name